### ISSUE TITLE
Add header_dropdown_menu module to core

### ISSUE THREAD

**Issue by izmeez** [backdrop/backdrop-issues#6718]
## Description of the need

The module https://github.com/backdrop-contrib/header_dropdown_menu is a great addition for Backdrop CMS and should be considered for inclusion in core.

```
This simple module enables site administrators to select "dropdown" menu style for the header block menu.

The dropdown menu is styled approximately like the Basis theme's Header block menu, but with added dropdown functionality. If you are using a different theme, you will most likely need to override the css rules found in header_dropdown_menu.css for best results.
```

**Comment by Egmund** [comment]
Yes indeed. I would also love to see dropdown included in core. That way I would not have had to look around and ask if it is possible.

**Comment by quicksketch** [comment]
I haven't heard of this module, but looking at it I can see why it would be useful.

When configuring a dedicated menu block, there is an option for Menu style as well as many options for how to display it:

![image](https://github.com/user-attachments/assets/5781ddbd-999a-405c-8110-42ba7ac29b93)

But there are no menu options when using the header block (which can also display a menu alongside logo, title, etc):

![image](https://github.com/user-attachments/assets/49e66da5-9186-4c0d-a9ee-4971b1c676ac)

I'm not sure if some or all of the menu settings should be copied into the header block as well.

I'd be curious to know what @argiepiano thinks about moving this into core, as he's the author of the module.
